gluten-core/src/main/scala/io/glutenproject/execution/SortExecTransformer.scala

```diff
@@ -52,123 +51,24 @@ case class SortExecTransformer(
 
   override def output: Seq[Attribute] = child.output
 
-  override def outputOrdering: Seq[SortOrder] = sortOrder
+  override def outputOrdering: Seq[SortOrder] = if (originalSortOrder.isEmpty) {
+    sortOrder
+  } else {
+    // After adding a pre-project to the sort, the expressions in sortOrder are replaced
+    // with attributes. To ensure that the downstream post-project can obtain the correct
+    // outputOrdering, it is necessary to restore the original sortOrder. Otherwise, when
+    // verifying whether the output attributes of the downstream are a subset of the
+    // references of sortOrder.child, it will not pass. Please check
+    // AliasAwareQueryOutputOrdering.outputOrdering in Spark-3.4.
+    originalSortOrder
```

Comment:
Detailed information can be viewed [here](https://github.com/oap-project/gluten/discussions/4308#discussioncomment-8036736).